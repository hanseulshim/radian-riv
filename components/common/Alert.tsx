import React from 'react'
interface Props {
  type: string
  message: string
}
export default function Alert({ type, message }: Props) {
  return (
    <div className={`alert alert-${type === 'error' ? 'danger' : 'success'}`}>
      <pre>{message}</pre>
    </div>
  )
}
