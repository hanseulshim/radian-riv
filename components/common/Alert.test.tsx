import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Alert from './Alert'

describe('Alert', () => {
  let component
  beforeAll(() => {
    const props = {
      type: 'success',
      message: 'Request Successful'
    }
    component = shallow(<Alert {...props} />)
  })
  it('has a div with class names', () => {
    expect(component.find('.alert')).toHaveLength(1)
    expect(component.find('.alert-success')).toHaveLength(1)
    expect(component.find('.alert-error')).toHaveLength(0)
  })
  it('has a <pre> tag with message', () => {
    expect(component.find('pre').text()).toEqual('Request Successful')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
