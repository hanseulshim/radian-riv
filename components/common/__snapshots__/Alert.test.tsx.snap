// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
<div
  className="alert alert-success"
>
  <pre>
    Request Successful
  </pre>
</div>
`;
